packages/components/managed-data-table/src/data-table-settings/column-manager/column-manager.js

```diff
@@ -0,0 +1,217 @@
+      const isSwap = result.source.droppableId === 'selected-columns-panel';
+
+      const items = isSwap
+        ? // remove the dragged item from its position if it is not coming from
+          // the hidden section (it is a swap)
+          [
+            ...props.selectedColumns.slice(0, result.source.index),
+            ...props.selectedColumns.slice(result.source.index + 1),
+          ]
+        : props.selectedColumns;
+
+      const columns = isSwap ? props.selectedColumns : props.availableColumns;
+      const draggedColumn = columns.find(
+        (col) => col.key === result.draggableId
+      );
+
+      // push the column in the new position
+      props.onUpdateColumns([
+        ...items.slice(0, result.destination.index),
+        draggedColumn,
+        ...items.slice(result.destination.index),
+      ]);
+    }
+  };
+
+  const handleDragStart = () => {
+    setIsDragging(true);
+  };
+
+  const hiddenColumns = differenceWith(
+    props.availableColumns,
+    props.selectedColumns,
+    (a, b) => a.key === b.key
+  );
+
+  return (
+    <CollapsibleMotion isDefaultClosed={false}>
+      {({ registerContentNode, containerStyles }) => (
+        <Container {...filterDataAttributes(props)}>
+          <Spacings.Stack scale="xs">
+            <HeaderContainer>
+              <Text.Headline as="h3" intlMessage={messages.title} />
+              <CloseButton onClick={props.onClose} size="medium" />
+            </HeaderContainer>
+            <Spacings.Inset scale="xs">
+              <div style={containerStyles}>
+                <div ref={registerContentNode}>
+                  <DragDropContext
+                    onDragEnd={handleDragEnd}
+                    onDragStart={handleDragStart}
+                  >
+                    <Spacings.Inline scale="m">
+                      <DroppableContainer isDragging={isDragging}>
+                        <Spacings.Stack>
+                          <Spacings.Inline scale="xs" alignItems="center">
+                            <EyeCrossedIcon size="medium" />
+                            <Text.Body isBold={true}>
+                              <FormattedMessage {...messages.hiddenColumns} />
+                            </Text.Body>
+                          </Spacings.Inline>
+                          {props.areHiddenColumnsSearchable && (
+                            <AsyncSelectInput
+                              id="hidden-columns-panel-search-input"
+                              {...(props.searchHiddenColumnsPlaceholder
+                                ? {
+                                    placeholder:
+                                      props.searchHiddenColumnsPlaceholder,
+                                  }
+                                : undefined)}
+                              cacheOptions={false}
+                              onChange={() => {
+                                // to avoid prop-types error
```

Comment:
But is the SelectInput then the right choice if selecting doesn't change anything?

Reply:
Being completely honest here, this is in the MC component, I didn't think about modifying it more than it was necessary. Actually yesterday I was looking at this and I noticed that I'm not using the props `areHiddenColumnsSearchable`,
`searchHiddenColumns` and `searchHiddenColumnsPlaceholder` on the parent, so currently all that logic would be useless. I think that I should check if it's being used on the MC and if that's the case I should allow to pass the props.

Reply:
Regarding the use of the `AsyncSelectInput`, as the code states `loadOptions` and `onInputChange` are used and `onChange` it's only included to avoid the warning of the required prop.